jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile

```diff
@@ -0,0 +1,184 @@
+                    sh 'echo SOCOK8S_DEVELOPER_MODE="False" >> jenkins.env'
+                    sh 'echo SOCOK8S_USE_VIRTUALENV="False" >> jenkins.env'
+                    sh 'echo DEPLOYMENT_MECHANISM="openstack" >> jenkins.env'
+                    sh 'echo ANSIBLE_STDOUT_CALLBACK="yaml" >> jenkins.env'
+                    sh 'echo USER="root" >> jenkins.env'
+
+                    // Start container
+                    env.CONTAINER_ID = sh(
+                        returnStdout: true,
+                        script: "podman run --env-file jenkins.env --rm -dt registry.opensuse.org/opensuse/leap/15.0/images/totest/images/opensuse-leap-15.0:current"
+                    ).trim()
+                }
+            }
+        }
+
+        stage('Install socok8s package') {
+            steps {
+                script {
+                    // FIXME(jhesketh): Add in Cloud:Rocky repo which has up-to-date openstacksdk
+                    //                  (details and alternative fix below)
+                    in_container('zypper addrepo http://download.opensuse.org/repositories/Cloud:/OpenStack:/Rocky/openSUSE_Leap_15.0/ "Cloud OpenStack Rocky"')
+                    // Install CA and socok8s
+                    in_container('zypper addrepo http://download.suse.de/ibs/SUSE:/CA/openSUSE_Leap_15.0/ "SUSE CA"')
+                    in_container('zypper addrepo https://download.opensuse.org/repositories/Cloud:/socok8s/openSUSE_Leap_15.0/ "SUSE OpenStack Cloud on Kubernetes Preview"')
+                    in_container('zypper --no-gpg-checks refresh')
+                    in_container('zypper --no-gpg-checks install -yl ca-certificates-suse socok8s')
+
+                    // FIXME(jhesketh): the openstacksdk library MUST be >=0.12.0 for ansible.
+                    //                  Install from pip for now until we resolve the proper deps
+                    //in_container('zypper --no-gpg-checks install -yl python3-pip')
+                    //in_container('pip3 install -U openstacksdk==0.27.0')
+                }
+            }
+        }
+
+        stage('Show environment information inside container') {
+            steps {
+                script {
+                    in_container('printenv')
+                }
+            }
+        }
+
+        stage('Copy in cloud config and sshkey') {
+            steps {
+                script {
+                    in_container('mkdir -p /root/.ssh')
+                    sh 'podman cp ~/.ssh/id_rsa $CONTAINER_ID:/root/.ssh/id_rsa'
+                    sh 'podman cp ~/.ssh/id_rsa.pub $CONTAINER_ID:/root/.ssh/id_rsa.pub'
+                    in_container('touch /root/.ssh/known_hosts')
+                    in_container('mkdir -p /etc/openstack')
+                    sh 'podman cp ~/.config/openstack/clouds.yaml $CONTAINER_ID:/etc/openstack/clouds.yaml'
+                }
+            }
+        }
+
+        stage('Create network') {
```

Comment:
Is it possible to move theses stages to a library or something? So we don't need to copy the steps into multiple Jenkinsfile's?

Reply:
Maybe, but because the jobs are in different repositories it becomes more complex. If we move the socok8s PR jenkins integration into this automation repo, it might be easier.

Given the simplicity of these steps I would be hesitant to add the complexity needed to make them a library. The real complexity inside those steps is already shared by being part of socok8s.

it is, however, a possible future improvement we could look at.

Reply:
Ah. True. I would move the Jenkinsfile to the socok8s repo. Otherwise, we can not test changes to that file with the current PR testing. And then we might be able to share stages between the jobs.

Reply:
The difference is that this isn't testing what is in git - this only tests what is in OBS. So it didn't make as much sense to put in the socok8s repo IMO. Granted, if we do look to turn these into a library it may make sense then.

Reply:
We test at least the Jenkinsfile itself if we move it to the socok8s git repo

Reply:
Yes, but these are (in my opinion) becoming different enough to lose any benefit of trying to turn them into a library. There is a bunch of extra logic going into the socok8s integration job to determine when to run which step. This job always knows what steps it wants to run and is a subset of the available ones.

You can test this file by triggering a manual run of the job in jenkins and setting the repo+branch of where to pull it from. That is how I have been running the tests on the aforementioned example runs.

Reply:
ok. I still don't like the amount of code we duplicate here but ok for now I guess.